Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -498,15 +512,39 @@ void AODProducerWorkflowDPL::addToFwdTracksTable(FwdTracksCursorType& fwdTracksC
     fwdCovInfo.rho1PtPhi = (Char_t)(128. * trackParamAtVertex.getCovariances()(2, 4) / (fwdCovInfo.sig1Pt * fwdCovInfo.sigPhi));
     fwdCovInfo.rho1PtTgl = (Char_t)(128. * trackParamAtVertex.getCovariances()(3, 4) / (fwdCovInfo.sig1Pt * fwdCovInfo.sigTgl));
 
-    const auto& rof = data.getMCHTracksROFRecords()[mMCHROFs[trackID.getIndex()]];
+    return true;
+  };
+
+  if (trackID.getSource() == GIndex::MCH) { // This is an MCH track
+    int mchTrackID = trackID.getIndex();
+    getMCHBitMap(mchTrackID);
+    if (!extrapMCHTrack(mchTrackID)) {
+      LOGF(warn, "Unable to extrapolate MCH track with ID %d! Dummy parameters will be used", mchTrackID);
+    }
+    fwdInfo.trackTypeId = o2::aod::fwdtrack::MCHStandaloneTrack;
+    const auto& rof = data.getMCHTracksROFRecords()[mMCHROFs[mchTrackID]];
     const auto bcWidth = 56; // do like in RecoContainerCreateTracksVariadic::createTracksVariadic
     fwdInfo.trackTime = (bcWidth / 2 + rof.getBCData().differenceInBC(mStartIR)) * o2::constants::lhc::LHCBunchSpacingNS;
     fwdInfo.trackTimeRes = o2::constants::lhc::LHCBunchSpacingNS * bcWidth / 2; // half interval
     errGaussian = false;
-
-  } else {
-    // This is a GlobalMuonTrack or a GlobalForwardTrack
-    const auto track = data.getGlobalFwdTrack(trackID);
+  } else if (trackID.getSource() == GIndex::MCHMID) { // This is an MCH-MID track
+    fwdInfo.trackTypeId = o2::aod::fwdtrack::MuonStandaloneTrack;
+    const auto& mchmidMatch = mchmidMatches[trackID.getIndex()];
+    const auto& mchTrackID = mchmidMatch.getMCHRef().getIndex();
```

Comment:
creating a reference on int is more expensive (8B instead of 4B + dereferencing) than simple copy, i.e.
```
auto mchmidMatch = mchmidMatches[trackID.getIndex()];
auto mchTrackID = mchmidMatch.getMCHRef().getIndex();
```
The same below.

Reply:
I agree for the indices, e.g. `mchTrackID`, but `mchmidMatch` is an object, so better getting a reference rather than copying it.

Reply:
Hi @shahor02, @pillot. Thank you for noticing the bug, I have removed the references now.

Reply:
well, it does not matter if it is an object or a simple type: being POD object its copy constructor is exactly the same as an assignment of int, so it is cheaper than the references. The reference would be make sense only if you want to modify it and preserve the modifications in the source.

Reply:
Reference from MCH-MID match is now removed as well